diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/utils/AstNodeUtils.kt

```diff
@@ -159,6 +159,11 @@ fun ASTNode.replaceWhiteSpaceText(beforeNode: ASTNode, text: String) {
 fun ASTNode.getFirstChildWithType(elementType: IElementType): ASTNode? =
         this.findChildByType(elementType)
 
+/**
+ * Checks if a function is anonymous
+ */
+fun ASTNode.isAnonymousFunction() = this.getIdentifierName() == null
```

Comment:
I would also add a check here that this method should only be called for `FUN` element types. We have such checks in a couple of other methods